### PR TITLE
Feature/3141/sort open api2

### PR DESCRIPTION
**Pull request by garyluu**
For #3141 

Originally tried to remove position, was not straightforward.  Only modifying  openapi.yaml

### CODE REVIEW

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TokenDAO.java  -52
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerFactory.java  -48
- dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java  -73
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicStateManager.java  -85
- dockstore-common/src/main/java/io/dockstore/common/LanguageHandlerHelper.java  -80
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java  -7
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java  -91
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PersistenceExceptionMapper.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Checksum.java  -46
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenTypeConverter.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceControlConverter.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java  -68
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java  -53
- dockstore-webservice/src/main/java/io/swagger/api/ToolsApiV1.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/User.java  -58
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApiService.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Token.java  -38
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/CustomWebApplicationException.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceUtilities.java  -81
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ServiceEntryDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Label.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ChecksumConverter.java  -55
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/AbstractDockstoreDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java  -88
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceFactory.java  -100
- dockstore-common/src/main/java/io/dockstore/common/LanguagePluginManager.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java  -46
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java  -86
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryStarredSerializer.java  -66
- dockstore-webservice/src/main/java/io/swagger/api/impl/ApiV2BetaVersionConverter.java  -74
- dockstore-common/src/main/java/io/dockstore/common/Utilities.java  -74
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java  -84
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/TransactionExceptionMapper.java  -91
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedToolResource.java  -31
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/FileDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java  -67
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/DescriptionSource.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java  -74
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/SourceControlResourceInterface.java  -62
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java  -72
- dockstore-webservice/src/main/java/io/openapi/api/impl/ToolClassesApiServiceImpl.java  -55
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java  -86
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthenticator.java  -61
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Event.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/FileFormat.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java  -40
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/CharsetResponseFilter.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java  -92
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  -28
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java  -88
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/URIHelper.java  -55
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZipSourceFileHelper.java  -4
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Service.java  -14
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java  -86
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/NotificationResource.java  -16
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ToolMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ToolDAO.java  -76
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventSearchType.java  -100
- dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthorizer.java  -20
- dockstore-webservice/src/main/java/io/swagger/api/impl/MetadataApiServiceImpl.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java  -58
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Alias.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/FileFormatHelper.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/NotificationDAO.java  -15
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/TRSListener.java  -83
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolClassesApiServiceImpl.java  -5
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/CollectionDAO.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TemplateHealthCheck.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/UserDAO.java  -26
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenType.java  -93
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Notification.java  -3
- dockstore-webservice/src/main/java/io/swagger/api/MetadataApiV1.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ObsoleteUrlFactory.java  -54
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java  -81
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java  -38
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguageSubclass.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsFactory.java  -69
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/VersionDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TOSVersion.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/BioWorkflowDAO.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ElasticSearchHealthCheck.java  -25
- dockstore-common/src/main/java/io/dockstore/common/SourceControl.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java  -82
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Organization.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/PrivacyPolicyVersion.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LabelDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/FileFormatDAO.java  -85
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java  -47
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ImageRegistryFactory.java  -47
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java  -84
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DAGHelper.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GoogleHelper.java  -18
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/RSSListener.java  -70
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java  -72
- dockstore-common/src/main/java/io/dockstore/common/EntryType.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApi.java  -22
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/tooltester/ToolTesterLogType.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java  -75
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ServiceResource.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/PublishRequest.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/doi/DOIGeneratorFactory.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/BioWorkflow.java  -69
- dockstore-common/src/main/java/io/dockstore/common/VersionTypeValidation.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/SitemapListener.java  -75
- dockstore-webservice/src/main/java/io/swagger/model/ExtendedFileWrapper.java  -33
- dockstore-common/src/main/java/io/dockstore/common/Registry.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java  -15
- dockstore-webservice/src/main/java/io/openapi/model/ExtendedFileWrapper.java  -41
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  -32
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/Role.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Image.java  -61
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java  -68
- dockstore-webservice/src/main/java/io/swagger/api/ToolClassesApiV1.java  -66
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/StateManagerMode.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Validation.java  -59
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryLabelHelper.java  -91
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/StateListenerInterface.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ToolTesterResource.java  -16
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TagDAO.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dag/ElementsDefinition.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  -83
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/MetadataResourceHelper.java  -33
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/VersionMetadata.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/OrganizationDAO.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java  -60
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=5060870&bid=16655959)

File: dockstore-webservice/scripts/sortOpenAPIYaml.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+yq r --tojson src/main/resources/openapi3/unsortedopenapi.yaml | yq r - -P > src/main/resources/openapi3/openapi.yaml
```
Comment (denis-yuen) [inline dockstore-webservice/scripts/sortOpenAPIYaml.sh]: Code comment, what does `-P` do?
Ideally, link to doc (might have missed it, but didn't find it in https://kislyuk.github.io/yq/ )

File: dockstore-webservice/scripts/sortOpenAPIYaml.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (denis-yuen) [inline dockstore-webservice/scripts/sortOpenAPIYaml.sh]: https://kvz.io/bash-best-practices.html (pipefail for example)

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,6 +20,12 @@ jobs:
           command: wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && sudo tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && rm dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
           environment:
             DOCKERIZE_VERSION: v0.3.0
+      - run: 
+          name: Install yq
```
Comment (coverbeck) [inline .circleci/config.yml]: TIL: yq (knew about jq)

File: dockstore-webservice/scripts/sortOpenAPIYaml.sh (hunk 1 above)
Comment (garyluu) [reply dockstore-webservice/scripts/sortOpenAPIYaml.sh]: https://github.com/mikefarah/yq

File: .circleci/config.yml (hunk 3 above)
Comment (garyluu) [reply .circleci/config.yml]: Will update primer

**Comment by garyluu** [comment]
LGTM  :cry:

**Pull request merged**